vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/OAuth2AwareInterceptor.java

```diff
@@ -36,29 +40,42 @@ public void handle(HttpContext<?> context) {
           .onSuccess(done -> context.next());
         break;
       case DISPATCH_RESPONSE:
-        processFailure(context);
+        processResponse(context);
         break;
       default:
         context.next();
         break;
     }
   }
 
-  private void processFailure(HttpContext<?> context) {
+  private void processResponse(HttpContext<?> context) {
```

Comment:
this whole section should be guarded by a `enable/disable` flag for users who wish to have full control and not rely on automatic retries.